core/src/main/scala/sttp/tapir/SchemaType.scala

```diff
@@ -105,7 +105,8 @@ object SchemaType {
   }
 
   case class SCoproduct[T](subtypes: List[Schema[_]], discriminator: Option[SDiscriminator])(
-      val subtypeSchema: T => Option[Schema[_]]
+      val subtypeSchema: T => Option[Schema[_]],
+      val applyValidation: (T, Map[SName, Schema[_]]) => List[ValidationError[_]] = ??? //FIXME: provide default
```

Comment:
ah, I see where the problem is. So far we've assumed that coproducts represent only subtyping relationships, that is the coproduct elements can be upcast to `T`. That's not true with `Either` - we need an unwrapping stage, similarly as with products. Maybe we need sth like `SProductField` ... let me think about it :)

Reply:
Maybe it would be enought if the `SCoproduct.subtypeSchema` function returned both a subtype and a value instead of just a subtype? I've updated the draft proposal to show what I mean.